File: go/structs.go
package main

type Player struct {
	Name        string
	Weapon      Weapon
	Health      int
	AttackDelay int
}

type Weapon struct {
	Name   string
	Damage int
	Delay  int
}

type Monster struct {
	Name   string
	Damage int
	Health int
}

File: c++/structs.cpp

#ifndef MYSTRUCT_H
#define MYSTRUCT_H

#include <string>

struct Weapon {
  std::string name;
  int damage;
  int delay;
};

struct Player {
  std::string name;
  int health;
  int attackDelay;
  Weapon weapon;
};

struct Monster {
  std::string name;
  int health;
  int damage;
};

#endif
